dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/processor/TaskDispatchProcessor.java

```diff
@@ -202,7 +198,12 @@ public void process(Channel channel, Command command) {
                         workerManager.getWaitSubmitQueueSize(),
                         taskExecutionContext.getTaskInstanceId());
                 workerMessageSender.sendMessageWithRetry(taskExecutionContext, masterAddress, CommandType.TASK_REJECT);
+                throw new TaskException("The task waiting queue is full, will reject this task");
             }
+        } catch (Exception ex) {
+            logger.error("Worker handle the dispatch task message failed, will remove the taskExecutionContext: {} from cache", taskExecutionContext);
+            TaskExecutionContextCacheManager.removeByTaskInstanceId(taskExecutionContext.getTaskInstanceId());
+            throw ex;
```

Comment:
Will the master dispatch fail if throw an exception here?

Reply:
Yes, the worker should send message to master be throw exception in the dispatch stage.